src/oatpp/core/data/stream/Stream.cpp

```diff
@@ -284,6 +284,82 @@ StreamType DefaultInitializedContext::getStreamType() const {
   return m_streamType;
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// BufferedInputStream
+
+v_io_size BufferedInputStream::flushBufferToStream(stream::WriteCallback *writeCallback) {
+  return writeBufferToStream(writeCallback, availableToRead());
+}
+
+async::CoroutineStarter BufferedInputStream::flushBufferToStreamAsync(const std::shared_ptr<data::stream::WriteCallback> &writeCallback) {
+  return writeBufferToStreamAsync(writeCallback, availableToRead());
+}
+
+v_io_size BufferedInputStream::writeBufferToStream(stream::WriteCallback *writeCallback, v_buff_size count) {
+  auto ioBuffer = buffer::IOBuffer::createShared();
```

Comment:
In simple API methods, there is no need for a shared object.
It might be better to use a regular `std::unique_ptr<char[]> buffer(new char[<size of IOBuffer>])`